src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -214,6 +214,7 @@ public async Task InitializeAsync(IEnumerable<FunctionMetadata> functions, Cance
 
             if (placeholderModeEnabled)
             {
+                _maxProcessCount = _webHostLanguageWorkerChannelManager.GetChannels(_workerRuntime).Count();
```

Comment:
I realize this avoids the uninitialized flow, but do we want to do something like we've discussed and put this behind a property to ensure initialization in all cases? 

Also, do we have a race between starting the channels at the web host level and the point where this count is performed?

Reply:
Yup, I'll make this a property.

Good point about a potential race here, any suggestions?

Reply:
Yes, race conditions are possible here - we have seen some variation of races here earlier as well - see

https://github.com/Azure/azure-functions-host/blob/2597f221d7643b24e206712ab9d8c3a01972b50d/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs#L405-L421

Instead of relying on the count - refactor the code above -get the count from webhostchannels - this waits on the Tasks that start in webhostchannelmanager